File: Scala/data-structures/src/main/scala/com/ckeyser/util/SortUtil.scala
package com.ckeyser.util

import Ordering.Implicits._

/**
  * Created by Chris on 9/28/2016.
  */
object SortUtil {
    def isSorted[A : Ordering](arr: Array[A]): Boolean = {
        for(i <- Range(0, arr.length - 1))
            if(arr(i) > arr(i + 1))
                false

        true
    }

    def swap[A](arr: Array[A], i: Int, j: Int): Unit = {
        val tmp = arr(i)
        arr(i) = arr(j)
        arr(j) = tmp
    }
}

File: Java/src/util/SortUtil.java
package util;

/**
 * Created by Chris Keyser on 10/3/2016.
 */
public class SortUtil {
    public static void swap(int[] arr, int i, int j) {
        int tmp = arr[i];
        arr[i] = arr[j];
        arr[j] = tmp;
    }

    /**
     * checks to see if an array is sorted
     */
    public static boolean isSorted(int[] arr) {
        for(int i = 0; i < arr.length - 2; i++) {
            if(arr[i] < arr[i + 1]) {
                return false;
            }
        }

        return true;
    }
}
